### ISSUE TITLE
null is not an object (evaluating 'x.store')

### ISSUE THREAD

**Issue by alexandrumic** [Nozbe/WatermelonDB#703]
I have a strange issue that happens only in release mode both on iOS and Android.
When building the app in release mode I encountered the following error:

`null is not an object (evaluating 'x.store')`

In the development mode is working ok so I can't follow the entire stracktrace. The error is the same both on iOS (Xcode logs) and Android (logcat).

Here is the full trace from Xcode:

```
2020-05-28 21:20:22.726773+0300 equinetapp[28114:178710] EXTaskService: Restoring tasks configuration: {
    mainApplication =     {
        tasks =         {
            "CREATE_EVENT_ENTRIES" =             {
                consumerClass = EXBackgroundFetchTaskConsumer;
                consumerVersion = 0;
                name = "CREATE_EVENT_ENTRIES";
                options =                 {
                    minimumInterval = 600;
                    startOnBoot = 1;
                    stopOnTerminate = 0;
                };
            };
            "FETCH_SCHEDULES" =             {
                consumerClass = EXBackgroundFetchTaskConsumer;
                consumerVersion = 0;
                name = "FETCH_SCHEDULES";
                options =                 {
                    minimumInterval = 900;
                    startOnBoot = 1;
                    stopOnTerminate = 0;
                };
            };
        };
    };
}
2020-05-28 21:20:22.946722+0300 equinetapp[28114:178710] Unbalanced calls to begin/end appearance transitions for <UIViewController: 0x7f9b76f07780>.
2020-05-28 21:20:23.544663+0300 equinetapp[28114:178710] Sentry Started -- Version: 4.4.3
2020-05-28 21:20:23.569 [error][tid:com.facebook.react.JavaScript] TypeError: null is not an object (evaluating 'x.store')

This error is located at:
    in A
    in Unknown
    in n
    in Unknown
    in RCTView
    in RCTView
    in Unknown
    in RCTView
    in h
2020-05-28 21:20:24.138077+0300 equinetapp[28114:179224] Deleted assets and updates from SQLite in 2.354026 ms
2020-05-28 21:20:24.138331+0300 equinetapp[28114:179230] Deleted 0 assets from disk in 0.002980 ms
2020-05-28 21:20:24.138444+0300 equinetapp[28114:179230] Retried deleting assets from disk in 0.002980 ms
2020-05-28 21:20:24.181 [error][tid:com.facebook.react.JavaScript] TypeError: null is not an object (evaluating 'x.store')

This error is located at:
    in A
    in Unknown
    in n
    in Unknown
    in RCTView
    in RCTView
    in Unknown
    in RCTView
    in h
2020-05-28 21:20:24.183 [fatal][tid:com.facebook.react.ExceptionsManagerQueue] Unhandled JS Exception: TypeError: null is not an object (evaluating 'x.store')

This error is located at:
    in A
    in Unknown
    in n
    in Unknown
    in RCTView
    in RCTView
    in Unknown
    in RCTView
    in h
2020-05-28 21:20:24.205051+0300 equinetapp[28114:179218] *** Terminating app due to uncaught exception 'RCTFatalException: Unhandled JS Exception: TypeError: null is not an object (evaluating 'x.store')

This error is located at:
    in A
    in Unknown
    in n
    in Unknown
    in RCTView
    in RCTView
    in Unknown
    in RCTView
    in h', reason: 'Unhandled JS Exception: TypeError: null is not an object (evaluating 'x.store')

This error is located at:
    in A
    in Unknown
    in n
    in Unknown
    in RCTView
    i..., stack:
A@728:1353
Or@97:41719
fi@97:48608
Ml@97:70036
Ul@97:67144
Ul@<null>:<null>
Pl@97:65839
Pl@<null>:<null>
<unknown>@97:25495
<unknown>@172:3915
sn@97:25442
cn@97:25377
kl@97:64831
la@97:87813
render@97:90592
<unknown>@347:541
run@339:1305
runApplication@339:2287
value@28:3685
<unknown>@28:841
value@28:2939
value@28:813
value@<null>:<null>
'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff23e39f0e __exceptionPreprocess + 350
	1   libobjc.A.dylib                     0x00007fff50ad79b2 objc_exception_throw + 48
	2   equinetapp                          0x0000000107e1f355 RCTFormatError + 0
	3   equinetapp                          0x0000000107e9448c -[RCTExceptionsManager reportFatalException:stack:exceptionId:] + 508
	4   CoreFoundation                      0x00007fff23e40e8c __invoking___ + 140
	5   CoreFoundation                      0x00007fff23e3e071 -[NSInvocation invoke] + 321
	6   CoreFoundation                      0x00007fff23e3e344 -[NSInvocation invokeWithTarget:] + 68
	7   equinetapp                          0x0000000107e52fba -[RCTModuleMethod invokeWithBridge:module:arguments:] + 578
	8   equinetapp                          0x0000000107e55244 _ZN8facebook5reactL11invokeInnerEP9RCTBridgeP13RCTModuleDatajRKN5folly7dynamicE + 246
	9   equinetapp                          0x0000000107e54fcc ___ZN8facebook5react15RCTNativeModule6invokeEjON5folly7dynamicEi_block_invoke + 78
	10  libdispatch.dylib                   0x00000001091d6f11 _dispatch_call_block_and_release + 12
	11  libdispatch.dylib                   0x00000001091d7e8e _dispatch_client_callout + 8
	12  libdispatch.dylib                   0x00000001091de6fd _dispatch_lane_serial_drain + 788
	13  libdispatch.dylib                   0x00000001091df28f _dispatch_lane_invoke + 422
	14  libdispatch.dylib                   0x00000001091eab65 _dispatch_workloop_worker_thread + 719
	15  libsystem_pthread.dylib             0x00007fff51b37a3d _pthread_wqthread + 290
	16  libsystem_pthread.dylib             0x00007fff51b36b77 start_wqthread + 15
)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

Any ideas? Thanks

**Comment by radex** [comment]
@alexandrumic I think it's in your code, I can't find any instances of `.store` in 🍉

**Comment by alexandrumic** [comment]
@radex It was happening after I added Watermelon, but I have searched for it and indeed, `x.store` is located in `ios/ProjectName/Supporting/app.bundle` for iOS. The app it's an Expo ejected app and these files are created on `expo publish` command before making a build. 

This is why I was thinking that maybe is related to Watermelon or something that Watermelon uses (SQLite or async storage). I will try to figure it out. 

Thank you
